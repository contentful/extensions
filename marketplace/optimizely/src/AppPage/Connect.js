import React from 'react';
import PropTypes from 'prop-types';
import { css } from 'emotion';
import tokens from '@contentful/forma-36-tokens';

import { Paragraph, Typography } from '@contentful/forma-36-react-components';

import ConnectButton from '../ConnectButton';

const styles = {
  spacing: css({
    margin: `${tokens.spacingM} 0`
  })
};

export default function Connect({ openAuth }) {
  return (
    <Typography>
      <Paragraph className={styles.spacing}>
        In order to see your experiments and connect them to Contentful content, we will need you to
        connect your Optimizely account by clicking on the button below. It will ask you to grant
        Contentful permissions to access your Optimizely experiments.
      </Paragraph>
      <ConnectButton openAuth={openAuth} />
    </Typography>
  );
}

Connect.propTypes = {
  openAuth: PropTypes.func.isRequired
};
