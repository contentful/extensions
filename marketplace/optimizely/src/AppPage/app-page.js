import React from 'react';
import PropTypes from 'prop-types';

import connectButton from './ConnectButton.svg';
const WINDOW_OPTS = 'left=150,top=150,width=700,height=700';

const url = `https://app.optimizely.com/oauth2/authorize
?client_id=15687650042
&redirect_uri=${encodeURIComponent('http://localhost:1234')}
&response_type=token
&scopes=all`;

function getAccessTokenFromHash(hash) {
  return (
    hash
      .slice(1)
      .split('&')[0]
      .split('=')[1] || null
  );
}

export default class AppPage extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      accessToken: '',
    };

    this.authWindow = () => {};

    this.listener = window.addEventListener(
      'message',
      event => {
        const { data, origin } = event;

        if (origin !== 'http://localhost:1234' || !data.hash) {
          return;
        }

        const accessToken = getAccessTokenFromHash(data.hash);

        if (typeof token !== 'string') {
          return;
        }

        this.setState({accessToken});
      },
      false
    );
  }

  openAuth = () => {
    window.open(url, '', WINDOW_OPTS);
  };

  render() {
    const { accessToken } = this.state;

    if (!accessToken) {
      return <button onClick={this.openAuth}>{connectButton}</button>;
    }

    return <div>App Page</div>;
  }
}
