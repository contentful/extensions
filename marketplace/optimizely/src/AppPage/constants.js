export const VARIATION_CONTAINER_ID = 'variationContainer';
