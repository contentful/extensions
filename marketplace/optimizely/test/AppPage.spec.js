import React from 'react';
import { cleanup, render, wait, fireEvent, act, configure } from '@testing-library/react';

import AppPage from '../src/AppPage';
import mockProps from './mockProps';
import projectData from './mockData/project.json';
import contentTypesData from './mockData/contentTypes.json';

const basicProps = {
  openAuth: () => {},
  accessToken: '',
  client: {
    getProjects: () => Promise.resolve(projectData)
  },
  sdk: mockProps.sdk
};

configure({ testIdAttribute: 'data-test-id' });

describe('AppPage', () => {
  afterEach(cleanup);
  // There is potentially a bug with @testing-library/react here. The order that these tests run
  // actually seems to matter. I tried calling `cleanup` in various ways but it didn't work.
  // This test must run before the other unfortunately.
  it('should render the Optimizely config and allow for adding content types', async () => {
    let configFunc = null;

    const platformAlpha = {
      app: {
        getParameters: jest.fn(() => Promise.resolve({ optimizelyProjectId: '123' })),
        onConfigure: jest.fn(fn => {
          configFunc = fn;
        }),
        setReady: jest.fn()
      }
    };

    const space = {
      getContentTypes: jest.fn(() => Promise.resolve(contentTypesData))
    };

    const sdk = { ...basicProps.sdk, platformAlpha, space };

    const props = { ...basicProps, accessToken: '123', sdk };

    const { container, getByTestId } = render(<AppPage {...props} />);

    await wait(() => {
      if (!configFunc) {
        throw '';
      }
    });

    await configFunc();
    expect(container).toMatchSnapshot();

    act(() => {
      fireEvent.click(getByTestId('add-content'));
    });

    // first we select the `select` element
    getByTestId('content-type-selector').click();
    // then we select the first `option` element
    getByTestId('content-type-selector').firstChild.click();

    expect(getByTestId('content-type-selector')).toMatchSnapshot();
  });

  it('should render the AppPage with Features and connect button', () => {
    const { container } = render(<AppPage {...basicProps} />);

    expect(container).toMatchSnapshot();
  });

  it('should render the AppPage loading when first connected to Optimizely', () => {
    const props = { ...basicProps, accessToken: '123' };

    const { container } = render(<AppPage {...props} />);

    expect(container).toMatchSnapshot();
  });
});
