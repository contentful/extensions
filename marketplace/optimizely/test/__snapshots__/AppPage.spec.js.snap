// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppPage should render the AppPage loading when first connected to Optimizely 1`] = `
<div>
  <div
    class="css-1a8f5ni"
  />
  <div
    class="css-jwwzlh"
  >
    <div>
      <div
        class="Typography__Typography___1ZUfE"
        data-test-id="cf-ui-text-container"
      >
        <h1
          class="Heading__Heading___83D3K f36-margin-bottom--m"
          data-test-id="cf-ui-heading"
        >
          Connect Optimizely
        </h1>
        <p
          class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
          data-test-id="cf-ui-paragraph"
        >
          You’re currently connected to Optimizely.
        </p>
        <p
          class="Paragraph__Paragraph___2aO-9 css-1ntki7d f36-margin-bottom--m"
          data-test-id="cf-ui-paragraph"
        >
          This access token is valid for 2 hours – after this you must reauthorize with Optimizely.
        </p>
      </div>
    </div>
    <hr
      class="css-14380qe"
    />
    <div>
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-id)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-id"
          >
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="35"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="59"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="80%"
              x="0"
              y="83"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation1"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
    <hr
      class="css-14380qe"
    />
    <div>
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-id)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-id"
          >
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="35"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="59"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="80%"
              x="0"
              y="83"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation1"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
  </div>
  <div
    class="css-fngmsh"
  >
    <svg
      height="60"
      width="60"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h60v60H0z"
          fill="#FFF"
        />
        <g
          fill-rule="nonzero"
        >
          <g
            fill="#0037FF"
          >
            <path
              d="M22.2 6.6a22 22 0 0 0-5.8 10 22 22 0 0 1 5.5-.8L15.7 22c0 1.4 0 2.8.3 4.1l10-10c1.4.2 2.6.6 3.9 1l1.2-1.6A9.4 9.4 0 1 1 43 30.1a21 21 0 0 1 1 3.8L34 44c1.4.3 2.8.4 4.2.4l6.2-6.2c0 1.8-.3 3.7-.7 5.5a22.1 22.1 0 1 0-21.4-37z"
            />
            <path
              d="M29.3 26.3a9.8 9.8 0 0 1-1-3.6l-5.6 5.7-4.4 4.3c.5 1 1.1 1.9 1.8 2.8l6.1-6.2 3-3zm8 5.4c-1.2-.1-2.4-.4-3.6-1l-3 3-6.1 6.2c.8.7 1.7 1.3 2.7 1.8l4.3-4.4 5.7-5.6z"
            />
          </g>
          <path
            d="M28.4 22.7c-.1-1.9.4-3.9 1.4-5.6-1.2-.4-2.5-.8-3.7-1l-10 10a21.3 21.3 0 0 0 2.2 6.7l10-10zm8.9 9l-10 10a22.2 22.2 0 0 0 6.6 2.3l10-10c-.2-1.3-.5-2.6-1-3.8a9.5 9.5 0 0 1-5.6 1.5zm6.3 11.9c.4-1.8.7-3.7.7-5.5l-6.2 6.2c1.8 0 3.6-.2 5.5-.7zm-27.2-27c-.5 1.7-.7 3.6-.7 5.4l6.2-6.2c-1.8 0-3.7.2-5.5.7zm17.3 14.2a9.6 9.6 0 0 1-4.4-4.5L20 35.5a24.6 24.6 0 0 0 4.4 4.4l9.2-9.1z"
            fill="#080738"
          />
          <path
            d="M16.4 16.5A22.1 22.1 0 0 0 0 37.6L15.7 22c0-1.8.2-3.7.7-5.5zm6 43.4a22.1 22.1 0 0 0 21.2-16.3 22 22 0 0 1-5.6.7L22.4 60zm-7-28.7c.5-.5 1-1 1.6-1.3-.4-1.2-.8-2.5-1-3.8L.4 41.8a22 22 0 0 0 2.3 6.6l10-10c-.1-2.6.8-5.3 2.8-7.2zM30.2 43a9.4 9.4 0 0 1-8.5 4.3l-10 10a22 22 0 0 0 6.7 2.3L33.9 44c-1.3-.2-2.5-.6-3.8-1zm-14.6 1.5a9.4 9.4 0 0 1-1.9-2.6l-9.2 9.2A24.6 24.6 0 0 0 9 55.6l9.2-9.2c-1-.5-1.9-1-2.6-1.9z"
            fill="#0037FF"
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`AppPage should render the AppPage with Features and connect button 1`] = `
<div>
  <div
    class="css-1a8f5ni"
  />
  <div
    class="css-jwwzlh"
  >
    <div>
      <div
        class="Typography__Typography___1ZUfE"
        data-test-id="cf-ui-text-container"
      >
        <h1
          class="Heading__Heading___83D3K f36-margin-bottom--m"
          data-test-id="cf-ui-heading"
        >
          Connect Optimizely
        </h1>
        <p
          class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
          data-test-id="cf-ui-paragraph"
        >
          You’re currently connected to Optimizely.
        </p>
        <p
          class="Paragraph__Paragraph___2aO-9 css-1ntki7d f36-margin-bottom--m"
          data-test-id="cf-ui-paragraph"
        >
          This access token is valid for 2 hours – after this you must reauthorize with Optimizely.
        </p>
      </div>
    </div>
    <hr
      class="css-14380qe"
    />
    <div>
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-id)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-id"
          >
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="35"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="59"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="80%"
              x="0"
              y="83"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation1"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
    <hr
      class="css-14380qe"
    />
    <div>
      <svg
        aria-label="Loading component..."
        class="SkeletonContainer__SkeletonContainer___23jiu"
        data-test-id="cf-ui-skeleton-form"
        height="100%"
        preserveAspectRatio="xMidYMid meet"
        role="img"
        width="100%"
      >
        <title>
          Loading component...
        </title>
        <rect
          clip-path="url(#cf-ui-skeleton-clip-id)"
          height="100%"
          style="fill: url(#cf-ui-skeleton-clip-gradient);"
          width="100%"
          x="0"
          y="0"
        />
        <defs>
          <clippath
            id="cf-ui-skeleton-clip-id"
          >
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="35"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="100%"
              x="0"
              y="59"
            />
            <rect
              height="16"
              rx="0"
              ry="0"
              width="80%"
              x="0"
              y="83"
            />
          </clippath>
          <lineargradient
            id="cf-ui-skeleton-clip-gradient"
          >
            <stop
              offset="0%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                dur="2s"
                id="animation1"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="50%"
              stop-color="#f7f9fa"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.25s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
            <stop
              offset="100%"
              stop-color="#e5ebed"
              stop-opacity="1"
            >
              <animate
                attributeName="stop-color"
                begin="animation1.begin+0.5s"
                dur="2s"
                repeatCount="indefinite"
                values="#e5ebed; #f7f9fa; #e5ebed"
              />
            </stop>
          </lineargradient>
        </defs>
      </svg>
    </div>
  </div>
  <div
    class="css-fngmsh"
  >
    <svg
      height="60"
      width="60"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h60v60H0z"
          fill="#FFF"
        />
        <g
          fill-rule="nonzero"
        >
          <g
            fill="#0037FF"
          >
            <path
              d="M22.2 6.6a22 22 0 0 0-5.8 10 22 22 0 0 1 5.5-.8L15.7 22c0 1.4 0 2.8.3 4.1l10-10c1.4.2 2.6.6 3.9 1l1.2-1.6A9.4 9.4 0 1 1 43 30.1a21 21 0 0 1 1 3.8L34 44c1.4.3 2.8.4 4.2.4l6.2-6.2c0 1.8-.3 3.7-.7 5.5a22.1 22.1 0 1 0-21.4-37z"
            />
            <path
              d="M29.3 26.3a9.8 9.8 0 0 1-1-3.6l-5.6 5.7-4.4 4.3c.5 1 1.1 1.9 1.8 2.8l6.1-6.2 3-3zm8 5.4c-1.2-.1-2.4-.4-3.6-1l-3 3-6.1 6.2c.8.7 1.7 1.3 2.7 1.8l4.3-4.4 5.7-5.6z"
            />
          </g>
          <path
            d="M28.4 22.7c-.1-1.9.4-3.9 1.4-5.6-1.2-.4-2.5-.8-3.7-1l-10 10a21.3 21.3 0 0 0 2.2 6.7l10-10zm8.9 9l-10 10a22.2 22.2 0 0 0 6.6 2.3l10-10c-.2-1.3-.5-2.6-1-3.8a9.5 9.5 0 0 1-5.6 1.5zm6.3 11.9c.4-1.8.7-3.7.7-5.5l-6.2 6.2c1.8 0 3.6-.2 5.5-.7zm-27.2-27c-.5 1.7-.7 3.6-.7 5.4l6.2-6.2c-1.8 0-3.7.2-5.5.7zm17.3 14.2a9.6 9.6 0 0 1-4.4-4.5L20 35.5a24.6 24.6 0 0 0 4.4 4.4l9.2-9.1z"
            fill="#080738"
          />
          <path
            d="M16.4 16.5A22.1 22.1 0 0 0 0 37.6L15.7 22c0-1.8.2-3.7.7-5.5zm6 43.4a22.1 22.1 0 0 0 21.2-16.3 22 22 0 0 1-5.6.7L22.4 60zm-7-28.7c.5-.5 1-1 1.6-1.3-.4-1.2-.8-2.5-1-3.8L.4 41.8a22 22 0 0 0 2.3 6.6l10-10c-.1-2.6.8-5.3 2.8-7.2zM30.2 43a9.4 9.4 0 0 1-8.5 4.3l-10 10a22 22 0 0 0 6.7 2.3L33.9 44c-1.3-.2-2.5-.6-3.8-1zm-14.6 1.5a9.4 9.4 0 0 1-1.9-2.6l-9.2 9.2A24.6 24.6 0 0 0 9 55.6l9.2-9.2c-1-.5-1.9-1-2.6-1.9z"
            fill="#0037FF"
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`AppPage should render the Optimizely config and allow for adding content types 1`] = `
<div>
  <div
    class="css-1a8f5ni"
  />
  <div
    class="css-jwwzlh"
  >
    <div>
      <div
        class="Typography__Typography___1ZUfE"
        data-test-id="cf-ui-text-container"
      >
        <h1
          class="Heading__Heading___83D3K f36-margin-bottom--m"
          data-test-id="cf-ui-heading"
        >
          Connect Optimizely
        </h1>
        <p
          class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
          data-test-id="cf-ui-paragraph"
        >
          You’re currently connected to Optimizely.
        </p>
        <p
          class="Paragraph__Paragraph___2aO-9 css-1ntki7d f36-margin-bottom--m"
          data-test-id="cf-ui-paragraph"
        >
          This access token is valid for 2 hours – after this you must reauthorize with Optimizely.
        </p>
      </div>
    </div>
    <hr
      class="css-14380qe"
    />
    <div
      class="Typography__Typography___1ZUfE"
      data-test-id="cf-ui-text-container"
    >
      <h1
        class="Heading__Heading___83D3K f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        Optimizely Project
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        Works only with Optimizely Full Stack projects
      </p>
      <div
        class="SelectField__SelectField___kbQlf css-15g2oxy"
        data-test-id="cf-ui-select-field"
        width="large"
      >
        <div
          class="SelectField__SelectField__label-wrapper___3jGwo"
        >
          <label
            class="FormLabel__FormLabel___3d6zQ"
            data-test-id="cf-ui-form-label"
            for="project"
          >
            Project
            <span
              class="FormLabel__FormLabel__required-text___3mvdm"
            >
              (
              required
              )
            </span>
          </label>
        </div>
        <div
          class="Select__Select__wrapper___2pBIz Select__Select--full___7ei7d"
        >
          <select
            aria-label="project"
            class="Select__Select___31Z46 a11y__focus-border--default___60AXp"
            data-test-id="cf-ui-select"
            id="project"
            name="project"
            required=""
          >
            <option
              data-test-id="cf-ui-select-option"
              value=""
            >
              Select Optimizely Project
            </option>
            <option
              data-test-id="cf-ui-select-option"
              value="57438502"
            >
              My Project
            </option>
            <option
              data-test-id="cf-ui-select-option"
              value="4567"
            >
              [Demo] - Docs tutorial
            </option>
          </select>
          <svg
            class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
            data-test-id="cf-ui-icon"
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
      </div>
    </div>
    <hr
      class="css-14380qe"
    />
    <div
      class="Typography__Typography___1ZUfE"
      data-test-id="cf-ui-text-container"
    >
      <h1
        class="Heading__Heading___83D3K f36-margin-bottom--m"
        data-test-id="cf-ui-heading"
      >
        Content types
      </h1>
      <p
        class="Paragraph__Paragraph___2aO-9 f36-margin-bottom--m"
        data-test-id="cf-ui-paragraph"
      >
        Select the content types for which you want to enable A/B testing
      </p>
      <button
        class="Button__Button___1ZfFj a11y__focus-border--default___60AXp Button__Button--muted___2Wair"
        data-test-id="add-content"
        type="button"
      >
        <span
          class="TabFocusTrap__TabFocusTrap___39Vty Button__Button__inner-wrapper___3qrNC"
          tabindex="-1"
        >
          <span
            class="Button__Button__label___3tcOj"
          >
            Add content type
          </span>
        </span>
      </button>
      <table
        cellpadding="0"
        cellspacing="0"
        class="css-1w2p18d Table__Table___2x3zV"
        data-test-id="cf-ui-table"
      >
        <tbody>
          <tr
            class="TableRow__TableRow___3kcNC"
            data-test-id="cf-ui-table-row"
          >
            <td
              align="left"
              class="TableCell__TableCell___Wou8a css-qd6c40"
              data-test-id="cf-ui-table-cell"
              style="top: 0px;"
            >
              <strong>
                Experiment
              </strong>
            </td>
            <td
              align="left"
              class="TableCell__TableCell___Wou8a css-qd6c40"
              data-test-id="cf-ui-table-cell"
              style="top: 0px;"
            >
              <div>
                Authors
                 
              </div>
            </td>
            <td
              align="left"
              class="TableCell__TableCell___Wou8a css-qd6c40"
              data-test-id="cf-ui-table-cell"
              style="top: 0px;"
            >
              <button
                class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 css-uoder TextLink__TextLink--primary___2Vc9F"
                data-test-id="cf-ui-text-link"
                type="button"
              >
                <span
                  class="TabFocusTrap__TabFocusTrap___39Vty"
                  tabindex="-1"
                >
                  Edit
                </span>
              </button>
              <button
                class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 css-uoder TextLink__TextLink--negative___3yhMk"
                data-test-id="cf-ui-text-link"
                type="button"
              >
                <span
                  class="TabFocusTrap__TabFocusTrap___39Vty"
                  tabindex="-1"
                >
                  Delete
                </span>
              </button>
            </td>
          </tr>
          <tr
            class="TableRow__TableRow___3kcNC"
            data-test-id="cf-ui-table-row"
          >
            <td
              align="left"
              class="TableCell__TableCell___Wou8a css-qd6c40"
              data-test-id="cf-ui-table-cell"
              style="top: 0px;"
            >
              <strong>
                Author
              </strong>
            </td>
            <td
              align="left"
              class="TableCell__TableCell___Wou8a css-qd6c40"
              data-test-id="cf-ui-table-cell"
              style="top: 0px;"
            >
              <div>
                Bio
                 
                <div
                  class="css-2k2f2j"
                >
                  <span
                    class="Tooltip__Tooltip__target-wrapper___Mtw42"
                  >
                    <svg
                      class="Icon__Icon___38Epv Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
                      data-test-id="cf-ui-icon"
                      height="18"
                      viewBox="0 0 24 24"
                      width="18"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M0 0h24v24H0z"
                        fill="none"
                      />
                      <path
                        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75l-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </td>
            <td
              align="left"
              class="TableCell__TableCell___Wou8a css-qd6c40"
              data-test-id="cf-ui-table-cell"
              style="top: 0px;"
            >
              <button
                class="TextLink__TextLink___1biUr a11y__focus-outline--default___2hwb1 css-uoder TextLink__TextLink--primary___2Vc9F"
                data-test-id="cf-ui-text-link"
                type="button"
              >
                <span
                  class="TabFocusTrap__TabFocusTrap___39Vty"
                  tabindex="-1"
                >
                  Edit
                </span>
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="css-fngmsh"
  >
    <svg
      height="60"
      width="60"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h60v60H0z"
          fill="#FFF"
        />
        <g
          fill-rule="nonzero"
        >
          <g
            fill="#0037FF"
          >
            <path
              d="M22.2 6.6a22 22 0 0 0-5.8 10 22 22 0 0 1 5.5-.8L15.7 22c0 1.4 0 2.8.3 4.1l10-10c1.4.2 2.6.6 3.9 1l1.2-1.6A9.4 9.4 0 1 1 43 30.1a21 21 0 0 1 1 3.8L34 44c1.4.3 2.8.4 4.2.4l6.2-6.2c0 1.8-.3 3.7-.7 5.5a22.1 22.1 0 1 0-21.4-37z"
            />
            <path
              d="M29.3 26.3a9.8 9.8 0 0 1-1-3.6l-5.6 5.7-4.4 4.3c.5 1 1.1 1.9 1.8 2.8l6.1-6.2 3-3zm8 5.4c-1.2-.1-2.4-.4-3.6-1l-3 3-6.1 6.2c.8.7 1.7 1.3 2.7 1.8l4.3-4.4 5.7-5.6z"
            />
          </g>
          <path
            d="M28.4 22.7c-.1-1.9.4-3.9 1.4-5.6-1.2-.4-2.5-.8-3.7-1l-10 10a21.3 21.3 0 0 0 2.2 6.7l10-10zm8.9 9l-10 10a22.2 22.2 0 0 0 6.6 2.3l10-10c-.2-1.3-.5-2.6-1-3.8a9.5 9.5 0 0 1-5.6 1.5zm6.3 11.9c.4-1.8.7-3.7.7-5.5l-6.2 6.2c1.8 0 3.6-.2 5.5-.7zm-27.2-27c-.5 1.7-.7 3.6-.7 5.4l6.2-6.2c-1.8 0-3.7.2-5.5.7zm17.3 14.2a9.6 9.6 0 0 1-4.4-4.5L20 35.5a24.6 24.6 0 0 0 4.4 4.4l9.2-9.1z"
            fill="#080738"
          />
          <path
            d="M16.4 16.5A22.1 22.1 0 0 0 0 37.6L15.7 22c0-1.8.2-3.7.7-5.5zm6 43.4a22.1 22.1 0 0 0 21.2-16.3 22 22 0 0 1-5.6.7L22.4 60zm-7-28.7c.5-.5 1-1 1.6-1.3-.4-1.2-.8-2.5-1-3.8L.4 41.8a22 22 0 0 0 2.3 6.6l10-10c-.1-2.6.8-5.3 2.8-7.2zM30.2 43a9.4 9.4 0 0 1-8.5 4.3l-10 10a22 22 0 0 0 6.7 2.3L33.9 44c-1.3-.2-2.5-.6-3.8-1zm-14.6 1.5a9.4 9.4 0 0 1-1.9-2.6l-9.2 9.2A24.6 24.6 0 0 0 9 55.6l9.2-9.2c-1-.5-1.9-1-2.6-1.9z"
            fill="#0037FF"
          />
        </g>
      </g>
    </svg>
  </div>
</div>
`;

exports[`AppPage should render the Optimizely config and allow for adding content types 2`] = `
<div
  class="SelectField__SelectField___kbQlf"
  data-test-id="content-type-selector"
>
  <div
    class="SelectField__SelectField__label-wrapper___3jGwo"
  >
    <label
      class="FormLabel__FormLabel___3d6zQ"
      data-test-id="cf-ui-form-label"
      for="content-types"
    >
      Content Type
      <span
        class="FormLabel__FormLabel__required-text___3mvdm"
      >
        (
        required
        )
      </span>
    </label>
  </div>
  <div
    class="Select__Select__wrapper___2pBIz Select__Select--medium___1Hmlq"
  >
    <select
      aria-label="content-types"
      class="Select__Select___31Z46 a11y__focus-border--default___60AXp"
      data-test-id="cf-ui-select"
      id="content-types"
      name="content-types"
      required=""
    >
      <option
        data-test-id="cf-ui-select-option"
        disabled=""
        value=""
      >
        Select content type
      </option>
      <option
        data-test-id="cf-ui-select-option"
        value="someOtherThing"
      >
        SomeOtherThing
      </option>
    </select>
    <svg
      class="Icon__Icon___38Epv Select__Select__icon___boRAZ Icon__Icon--small___1yGZK Icon__Icon--muted___3egnD"
      data-test-id="cf-ui-icon"
      height="24"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
</div>
`;
